portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,36 +44,51 @@
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  // A list of all comments.
-  ArrayList<Comment> commentsList = new ArrayList<>();
-
   /** Processes GET requests for "/comments" and returns a list of comments in JSON format. */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonList();
+    // Create new Query for Commen objects.
+    Query query = new Query("Comment");
 
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
-  }
+    // Get access to dataStore.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
 
-  private String convertToJsonList() {
+    // Get list of comments.
+    List<Comment> commentsList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("text");
+      int rating = ((Long) entity.getProperty("rating")).intValue();
+      Comment comment = new Comment(name, text, rating);
+      
+      commentsList.add(comment);
+    }
+
+    // Convert list of comments to JSON format.
     Gson gson = new Gson();
     String json = gson.toJson(commentsList);
-    return json;
+
+    // Return response to the request.
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+    return;
   }
 
   /** Processes POST request by storing received commentary. */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
```

Comment:
nit: for consistency, code blocks such as functions should not start with an empty line.